algorithm.py

```diff
@@ -20,11 +24,9 @@
 
 Tutorial
 --------
-
-This module provides an abstraction for implementing arbitrary algorithms as a
-list of functions that operate on a shared state dictionary. Algorithms defined
-this way are easy to arbitrarily modify at run time, and they provide cascading
-exception handling.
+The algorithm in this module is a list of functions that are
+executed in defined order, the return values of which update the
+algorithm's run state in a `state` dictionary.
```

Comment:
Because `The model` is too abstract. I tried to make text less academic, because the module provides a fixed algorithm - no need to say `model of algorithm`. Perhaps a better wording would be "The module provides a fixed algorithm defined as a list of functions that are executed in sequential order. The return value from each function updates state of algorithm stored in `state` dictionary.".